birdclef/utils.py

```diff
@@ -56,17 +56,47 @@ def load_audio(input_path: Path, offset: float, duration: int = 7, sr: int = 320
     return np.resize(np.moveaxis(y_trunc, -1, 0), length)
 
 
-def slice_seconds(data, sample_rate, seconds=5, pad_seconds=0):
+def slice_seconds(
+    data, sample_rate, seconds=5, pad_seconds=0, remainder_padding_type="center"
+):
+    valid_padding_types = ["center", "right"]
+    if remainder_padding_type not in valid_padding_types:
+        raise ValueError(
+            f"remainder_padding_type must be one of: {valid_padding_types}"
+        )
+
     # return 2d array of the original data
     n = len(data)
     k = sample_rate * seconds
     pad = sample_rate * pad_seconds
+    step = k + pad
+
     indexes = np.array(
-        [np.arange(i, i + k + pad) for i in range(0, n, k) if i + k + pad <= n]
+        [np.arange(i, i + step) for i in range(0, n, k) if i + step <= n]
     ).astype(int)
     indexed = data[indexes]
     if indexed.shape[0] == 0:
         return []
+
+    # if data doesn't divide evenly into the slice duration, add padding to the remainder slice
+    remainder = n % step
+    if remainder:
+        start_index = n // step * step
+        remainder_indexes = np.arange(start_index, start_index + remainder)
+        remainder_slice = data[remainder_indexes]
+        padding_size = step - remainder
+        if (
+            remainder_padding_type == "center"
+        ):  # centers the padding like [0, 0, 1, 0, 0]
```

Comment:
So a minor miscommunication here -- I was actually thinking that we would pad the entire audio file instead of just the last slice. I do appreciate the implementation though.